ns_pipelines/participant_demographics/run.py

```diff
@@ -0,0 +1,88 @@
+""" Extract participant demographics from HTML files. """
+import os
+
+from publang.extract import extract_from_text
+from openai import OpenAI
+import logging
+
+from . import prompts
+from .clean import clean_prediction
+
+from ns_pipelines.pipeline import IndependentPipeline
+
+
+def extract(extraction_model, extraction_client, text, prompt_set='', **extract_kwargs):
+    extract_kwargs.pop('search_query', None)
+
+    # Extract
+    predictions = extract_from_text(
+        text,
+        model=extraction_model,
+        client=extraction_client,
+        **extract_kwargs
+    )
+
+    if not predictions:
+        logging.warning("No predictions found.")
+        return None, None
+
+    clean_preds = clean_prediction(predictions)
+
+    return predictions, clean_preds
+
+
+def _load_client(model_name):
+    if 'gpt' in model_name:
+        client = OpenAI(api_key=os.getenv('OPENAI_API_KEY'))
+
+    else:
```

Comment:
In principle we can run this using other API keys and hence other model names, so perhaps let's not worry about validation here